cli/upload.py

```diff
@@ -307,20 +309,37 @@ def _compose_file_mapping(upload_info: api.UploadInfo, xlsx: str):
             source_path = os.path.join(xlsx_dir, source_path)
 
             if not os.path.isfile(source_path):
-                missing_files.append(source_path)
+                if source_path in upload_info.optional_files:
+                    skipping_files.append(gcs_uri)
+                    continue
+                else:
+                    missing_files.append(source_path)
 
         # gsutil treats brackets in a gs-uri as a character set
         # see https://cloud.google.com/storage/docs/gsutil/addlhelp/WildcardNames#other-wildcard-characters
         # this replaces opening with a generic wildcard, which should map to only a single file
         elif "[" in source_path and "]" in source_path:
             source_path = source_path.replace("[", "?")
+            sub = subprocess.run(
+                ["gsutil", "ls", f"'{source_path}'"], capture_output=True
+            )
```

Comment:
Looks like this code only checks if a file exists in GCS if it contains square brackets in the URI. Shouldn't we be checking all GCS URIs?

Reply:
fixed in next commit